File: app/src/main/java/com/anthonymandra/rawdroid/data/FolderEntity.java
package com.anthonymandra.rawdroid.data;

import android.arch.persistence.room.Entity;

@Entity(tableName = "image_parent")
public class FolderEntity extends PathEntity
{
	FolderEntity(String documentUri, Long id, String path, int depth, Long parent)
	{
		super(id, path, depth, parent);
		this.documentUri = documentUri;
	}

	public String documentUri;   // TODO: This might not be necessary, prolly should have uri though

	@Override
	public boolean equals(Object obj)
	{
		if (obj instanceof FolderEntity)
		{
			FolderEntity compare = (FolderEntity) obj;
			return super.equals(obj) && documentUri.equals(compare.documentUri);
		}
		return false;
	}
}

File: app/src/main/java/com/anthonymandra/rawdroid/data/FolderEntity.kt
package com.anthonymandra.rawdroid.data

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "image_parent",
        indices = [(Index(value = ["documentUri"], unique = true))])
data class FolderEntity @JvmOverloads constructor(
        var documentUri: String = "",
        var visible: Boolean = true,
        var excluded: Boolean = false,
        @PrimaryKey(autoGenerate = true)
        var id: Long = 0)

